### DISCUSSION TITLE
Explicit task

### DISCUSSION THREAD

**Discussion by pitermarx** [cake-build/cake#3380] [Ideas]
I keep repeating myself creating tasks that can only be run explicitly or when an argument is passed
Is this something that the community finds useful?
I tried to make a PR but I don't think it's possible unless the cake context has info about the target

![image](https://user-images.githubusercontent.com/3081298/119810645-20f6e280-bede-11eb-9ad9-49f50a4d8485.png)

**Comment by pitermarx** [comment]
Just realized 2 things
1 - I could just c.Arguments.GetArguments("Target") (doh 🤦‍♂️)
2 - Cake does not have the concept of "target". That's something you define on the script

**Reply by devlead** [reply]
Setup context does though, so you could have a build data record/class that you pass that to and use in WithCriteria if it should be executed or not.

https://cakebuild.net/api/Cake.Core/ISetupContext/

**Reply by pitermarx** [reply]
right now I'm doing this
![image](https://user-images.githubusercontent.com/3081298/119814907-dcba1100-bee2-11eb-89e2-7146235ec3c0.png)

**Comment by gitfool** [comment]
Another way to handle this would be to extend Cake to support specifying multiple targets, combined with exclusive mode you would have the same result:
```sh
cake --targets clean,build --exclusive
```
